workers/unity/Packages/com.improbable.gdk.querybasedinterest/Constraint.cs

```diff
@@ -0,0 +1,139 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    public class Constraint
+    {
+        public static ComponentInterest.QueryConstraint Sphere(double radius, Vector3 center)
+        {
+            return new ComponentInterest.QueryConstraint
+            {
+                AndConstraint = new List<ComponentInterest.QueryConstraint>(),
```

Comment:
Why do and and or need to tbe set here but not everything else? I thought you are only ever supposed to set one.

Reply:
Yup you're only meant to set one of the constraints. However, every other constraint is an Option<> (so they can be left out) but And/Or are List<>. It's something I want to ask runtime about.

Reply:
I think this is something we should deal with at serialisation. Otherwise we should have a private static `Empty` list somewhere that we set things too, if we know nothing will mutate it.